ansible/vars/topo_ciscovs-7nodes.yml

```diff
@@ -0,0 +1,218 @@
+        - 30
+      vm_offset: 2
+    P3:
+      vlans:
+        - 31
+      vm_offset: 3
+    P2:
+      vlans:
+        - 16
+      vm_offset: 4
+    P4:
+      vlans:
+        - 17
+      vm_offset: 5
+  VM_LINKs:
+    PE1P3:
+      start_vm_offset: 0
+      start_vm_port_idx: 1
+      end_vm_offset: 3
+      end_vm_port_idx: 1
+    PE2P3:
+      start_vm_offset: 1
+      start_vm_port_idx: 2
+      end_vm_offset: 3
+      end_vm_port_idx: 2
+    P3P2:
+      start_vm_offset: 3
+      start_vm_port_idx: 3
+      end_vm_offset: 4
+      end_vm_port_idx: 1
+    P3P4:
+      start_vm_offset: 3
+      start_vm_port_idx: 4
+      end_vm_offset: 5
+      end_vm_port_idx: 1
+    P2P4:
+      start_vm_offset: 4
+      start_vm_port_idx: 2
+      end_vm_offset: 5
+      end_vm_port_idx: 2
+  OVS_LINKs:
+    P2PE3:
+      vlans:
+        - 39
+      start_vm_offset: 4
+      start_vm_port_idx: 3
+      end_vm_offset: 2
+      end_vm_port_idx: 1
+    P4PE3:
+      vlans:
+        - 40
+      start_vm_offset: 5
+      start_vm_port_idx: 3
+      end_vm_offset: 2
+      end_vm_port_idx: 3
+  DUT:
+    vlan_configs:
+      default_vlan_config: one_vlan_a
+      one_vlan_a:
+        Vlan1000:
+          id: 1000
+          intfs: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 21, 22, 23, 24]
+          prefix: 192.168.0.1/21
+          prefix_v6: fc02:1000::1/64
+          tag: 1000
+      two_vlan_a:
+        Vlan100:
+          id: 100
+          intfs: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
+          prefix: 192.168.0.1/22
+          prefix_v6: fc02:100::1/64
+          tag: 100
+        Vlan200:
+          id: 200
+          intfs: [13, 14, 15, 21, 22, 23, 24]
+          prefix: 192.168.4.1/22
+          prefix_v6: fc02:200::1/64
+          tag: 200
+      four_vlan_a:
+        Vlan1000:
+          id: 1000
+          intfs: [1, 2, 3, 4, 5, 6]
+          prefix: 192.168.0.1/23
+          prefix_v6: fc02:400::1/64
+          tag: 1000
+        Vlan2000:
+          id: 2000
+          intfs: [7, 8, 9, 10, 11, 12]
+          prefix: 192.168.2.1/23
+          prefix_v6: fc02:401::1/64
+          tag: 2000
+        Vlan3000:
+          id: 3000
+          intfs: [13, 14, 15]
+          prefix: 192.168.4.1/23
+          prefix_v6: fc02:402::1/64
+          tag: 3000
+        Vlan4000:
+          id: 4000
+          intfs: [21, 22, 23, 24]
+          prefix: 192.168.6.1/23
+          prefix_v6: fc02:403::1/64
+          tag: 4000
+
+configuration_properties:
+  common:
+    dut_asn: 65100
+    dut_type: ToRRouter
+    swrole: leaf
+    nhipv4: 10.10.246.254
+    nhipv6: FC0A::FF
+    podset_number: 200
+    tor_number: 16
+    tor_subnet_number: 2
+    max_tor_subnet_number: 16
+    tor_subnet_size: 128
+    spine_asn: 65534
+    leaf_asn_start: 64600
+    tor_asn_start: 65500
+    failure_rate: 0
+
+dut_cfg_file_loc : vars/configdb_jsons/7nodes_cisco/P1.json
+
+configuration:
+  PE1:
+    cfg_file_loc : vars/configdb_jsons/7nodes_cisco/PE1.json
```

Comment:
Embedding path information in a logical topology definition file does not really make sense to me. How about define the path information in separate variable files?

Reply:
current topo file provides initial configurations for each node, although they are follow minigraph or generated vtysh configuration directly. 

This cfg file provides the initial configuration in config db format. They serve the same purpose, right?

Reply:
The current topo file logically describes the testbed topology. Including information of what type of nodes to be used and how they should be configured. The initial configuration file is more like some configuration profile or template. 

It would be better to use some profile/template name here in topology definition file. Embedding path information here does not really make sense to me.

How about change this to something like below:
```
init_cfg_profile: 7nodes_cisco_PE1
```

Then use a separate file to map the profile name to actual cfg file location, for example: `init_cfg_profiles.yml`
```
7nodes_cisco_PE1: vars/config_db_jsons/7nodes_cisco/PE1.json
...
```

Reply:
Done with suggested changes

https://github.com/sonic-net/sonic-mgmt/pull/13785/commits/a702fde13ee894b42fce524315d628455f772db4